### PR TITLE
Jamesweb dev

### PR DESCRIPTION
**Pull request by jamesweb-ms**
Dev deployment experience

### CODE REVIEW

File: EventProcessor/EventProcessor.WorkerRole/Processors/ActionEventProcessor.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ public void Stop()
 
                 _eventProcessorHost = new EventProcessorHost(
                     hostName,
-                    eventHubPath,
+                    eventHubPath.ToLower(),
```
Comment (keithcraig) [inline EventProcessor/EventProcessor.WorkerRole/Processors/ActionEventProcessor.cs]: Should this be .ToLowerInvariant()?

File: Common/Deployment/DeploymentLib.ps1 [hunk 2]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
-    {
-        Add-AzureAccount
-        $locations = Execute-Command -Command ("Get-AzureLocation")
-    }
+    param ([Parameter(Mandatory=$true)][string]$location)
+    $locations = @("East US", "North Europe", "East Asia")
     foreach ($loc in $locations)
     {
-        if ($loc.Name -eq $location)
+        if ($loc.Replace(' ', '').Tolower() -eq $location.Replace(' ', '').Tolower())
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should these be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 3]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
-	write-verbose "GetType.Name $($exc.Exception.GetType().Name)"
-    if ($exc.Exception.GetType().Name -eq "ResourceNotFoundException")
+    # Return name for existing resource if exists
+    $resources = Get-AzureResource -ResourceGroupName $resourceGroupName -ResourceType $resourceType -OutputObjectFormat New
+    if ($resources -ne $null)
     {
-        return $true
+        foreach($resource in $resources)
+        {
+            if ($resource.Name.ToLower().StartsWith($resourceBaseName.ToLower()))
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should these be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 4]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
-	write-verbose "Messsage $($exc.Exception.Message)"
-	if ($exc.Exception.Message.StartsWith("ResourceNotFound"))
-	{
-		return $true
-	}
 
-    if ($exc.Exception.GetType().Name -ne "ServiceManagementClientException")
+    # Generate a unique name
+    $resourceUrl = " "
+    switch ($resourceType.ToLower())
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should these be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 5]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
+function GetAzureDocumentDbName()
 {
     Param(
-	    [Parameter(Mandatory=$true,Position=0)] [string] $serviceBaseName,
-        [Parameter(Mandatory=$true,Position=1)] [string] $serviceBusNamePrefix,
-        [Parameter(Mandatory=$true,Position=2)] [string] $eventHubPath
+        [Parameter(Mandatory=$true,Position=0)] [string] $baseName,
+        [Parameter(Mandatory=$true,Position=1)] [string] $resourceGroupName
     )
+    return ValidateResourceName $baseName.ToLower() Microsoft.DocumentDb/databaseAccounts $resourceGroupName
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should these be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 6]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
+function UploadFile()
+{
+    Param(
+        [Parameter(Mandatory=$true,Position=0)] [string] $filePath,
+        [Parameter(Mandatory=$true,Position=1)] [string] $storageAccountName,
+        [Parameter(Mandatory=$true,Position=2)] [string] $resourceGroupName,
+        [Parameter(Mandatory=$true,Position=3)] [string] $containerName
+    )
+    $maxSleep = 60
+    $containerName = $containerName.ToLower()
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should these be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 7]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
-			return $store.StorageAccountName
-		}
-		if ($max-- -eq 0)
-		{
-			Write-Error -Message "Unable to create storage account for prefix $storageNamePrefix"
-		}
-		$name = "{0}{1:x}" -f $storageNamePrefix, (Get-Date).Millisecond
-	}
-	exit 1
+    $storageAccountName = ValidateResourceName $storageBaseName.ToLower() Microsoft.Storage/storageAccounts $resourceGroupName
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should this be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 8]
```diff
@@ -90,348 +98,242 @@ function GetAuthenticationResult()
...
+    Param(
+        [Parameter(Mandatory=$true,Position=0)] [string] $filePath,
+        [Parameter(Mandatory=$true,Position=1)] [string] $storageAccountName,
+        [Parameter(Mandatory=$true,Position=2)] [string] $resourceGroupName,
+        [Parameter(Mandatory=$true,Position=3)] [string] $containerName
+    )
+    $maxSleep = 60
+    $containerName = $containerName.ToLower()
+    $file = Get-Item -Path $filePath
+    $fileName = $file.Name.ToLower()
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Should this be `.ToLowerInvariant()`?

File: Common/Deployment/DeploymentLib.ps1 [hunk 9]
```diff
@@ -1242,12 +687,11 @@ if ((Get-Module | where {$_.Name -match "Azure"}) -eq $Null)
...
-	}
-	else
-	{
-	    Write-Error -Category ObjectNotFound -Message "Unable to find Azure.psd1 modules. Please install Azure Powershell 2.5.1 or later"
-        exit 1
-	}
+    }
+    else
+    {
+        thow "Unable to find Azure.psd1 modules. Please install Azure Powershell 2.5.1 or later"
```
Comment (keithcraig) [inline Common/Deployment/DeploymentLib.ps1]: Appears to be a typo of `throw`

File: WebJobHost/Web.config [hunk 10]
```diff
@@ -1,76 +1,83 @@
...
+    <add key="SolutionName" value="Remote Monitoring Solution"/>
+    <add key="TelemetryStoreContainerName" value="devicetelemetry"/>
+    <add key="TelemetryDataPrefix" value="devicetelemetry/"/>
+    <add key="TelemetrySummaryPrefix" value="devicetelemetry-summary/"/>
+    <add key="AlertsStoreContainerName" value="rulesoutput"/>
+    <add key="DeviceAlertsDataPrefix" value=""/>
+    <add key="MapApiQueryKey" value="config:MapApiQueryKey"/>
   </appSettings>
-
   <system.web>
```
Comment (keithcraig) [inline WebJobHost/Web.config]: This section appears to be a duplicate of the `<system.web>` section on lines 15 - 18 above.

Comment (keithcraig) [inline WebJobHost/Web.config]: The `<system.web>` section on lines 45 - 51 appears to be a duplicate of the `<system.web>` section on lines 15 - 18.

File: EventProcessor/EventProcessor.WorkerRole/Processors/ActionEventProcessor.cs (hunk 1 above)
Comment (jamesweb-ms) [inline EventProcessor/EventProcessor.WorkerRole/Processors/ActionEventProcessor.cs]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 2 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 3 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 4 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 5 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 6 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 7 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 8 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: Fixed

File: Common/Deployment/DeploymentLib.ps1 (hunk 9 above)
Comment (jamesweb-ms) [inline Common/Deployment/DeploymentLib.ps1]: fixed

File: WebJobHost/Web.config (hunk 10 above)
Comment (jamesweb-ms) [inline WebJobHost/Web.config]: fixed

**Pull request merged**